pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -457,7 +457,123 @@ func Test_server_Search(t *testing.T) {
 		}
 		return nil
 	}
+
+	/*
+		Search test cases:
+		- Equivalence Class Testing
+			- uint8, float32 common
+				- case 1: vector success
+				- case 2: different dimension fail
+		- Boundary Value Testing
+		  	- uint8, float32 common
+				- case 1: min value vector success
+				- case 2: max value vector success
+				- case 3: 0 length vector fail?
+				- case 4: max length vector fail?
+				- case 5: nil vector fail
+			- uint8
+			- float32
+				- case 6: 0 value vector success
+				- case 7: NaN value vector fail?
+		- Decision Table Testing
+			- uint8, float32 common
+		    | same vector       | false | false | false | true | true | true |
+			| inserted          |     5 |    10 |    20 |    5 |   10 |   20 |
+			| Search_Config.Num |    10 |    10 |    10 |   10 |   10 |   10 |
+				- case 1: inserted 5 different vector, Num is 10 success
+				- case 2: inserted 10 different vector, Num is 10 success
+				- case 3: inserted 20 different vector, Num is 10 success
+				- case 4: inserted 5 same vector, Num is 10 success?
+				- case 5: inserted 10 same vector, Num is 10 success?
+				- case 6: inserted 20 same vector, Num is 10 success?
```

Comment:
I think there are some cases that the vector is too far from the search vector and cannot be searched, do you think we need to consider these case also?

Reply:
If my memory is correct, in the initial phase of NGT each node has dense connections.
So we need not consider the cases.
However, in the pkg decision table test, I don't know which pattern is better for inserting a few vectors or many vectors like below.

```
- case 1: insert 1000 different vectors, Num is 10 success
- case 2: insert 10000 different vectors, Num is 10 success
- case 3: insert 100000 different vectors, Num is 10 success
- case 4: insert 1000 same vectors, Num is 10 success(?)
- case 5: insert 10000 same vectors, Num is 10 success(?)
- case 6: insert 100000 same vectors, Num is 10 success(?)
```

In these cases, it takes more time than above.

Reply:
umm... I think many vectors in a case is fine as it is a more realistic use case of NGT.
@kmrmt @vankichi  what do you think?

Reply:
I think the number of vectors to insert should be as large as possible.